lib/job-iteration/iteration.rb

```diff
@@ -189,7 +189,7 @@ def iterate_with_enumerator(enumerator, arguments)
         # Deferred until 2.0.0
         # assert_valid_cursor!(cursor_from_enumerator)
 
-        tags = instrumentation_tags.merge(cursor_position: cursor_from_enumerator)
+        tags = instrumentation_tags.merge(cursor_position: cursor_from_enumerator).deep_dup
```

Comment:
To be honest I'm not entirely sure why this is happening, but the added test in `test/unit/iteration_test.rb` fails otherwise: the notification payload is 12 times `{:job_class=>"JobIterationTest::JobWithNestedEnumerator", :cursor_position=>[2, nil]}`   instead of the expected increasing nested cursor 🤔

Reply:
It seems to hang on to `@cursors` yielded from the nested enumerator.
```
irb(#<JobIterationTest:0x00000001066631c0>):003:0> events.map { |e| e[:cursor_position].object_id }
=> [243980, 243980, 243980, 243980, 243980, 243980, 243980, 243980, 243980, 243980, 243980, 243980]
```
`cursor_from_enumerator.dup` instead of the tags object works too, I'll push that up in a sec.